src/InstrumentationEngine.Attach/EngineConfiguration.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Xml.Serialization;
+
+namespace Microsoft.InstrumentationEngine
+{
+    public class InstrumentationEngineConfiguration
+    {
+        public InstrumentationEngine? InstrumentationEngine { get; set; }
+
+        public InstrumentationMethod[] InstrumentationMethods { get; set; }
```

Comment:
>InstrumentationMethod[] [](start = 15, length = 23)

Make a new intermediate class (e.g. InstrumentationMethodsCollection) that has an object[] Items property on it. Decorate that property with the XmlElement attribute with the name "AddInstrumentationMethod" and points to the AddInstrumentationMethod type.

Reply:
Generating from schema avoids this.

---
In reply to: [387843676](https://github.com/microsoft/CLRInstrumentationEngine/pull/220#discussion_r387843676) [](ancestors = 387843676)